### PR TITLE
[pytree] Allow register_pytree_node to take in 5 inputs

### PR DESCRIPTION
**Pull request by angelayi**
Summary: This is currently breaking internal old torch.packages from when _register_pytree_node took 5 inputs.

Test Plan: CI

Differential Revision: D48834742

### CODE REVIEW

**Review by zhxchen17 — APPROVED** [review]

File: torch/utils/_pytree.py [hunk 1]
```diff
@@ -72,7 +72,6 @@ def _register_pytree_node(
     typ: Any,
     flatten_fn: FlattenFunc,
     unflatten_fn: UnflattenFunc,
-    *,
```
Comment (zou3519) [inline torch/utils/_pytree.py:75]: Are we ever able to undo this change?

Comment (angelayi) [reply torch/utils/_pytree.py:75]: Yes, but whenever torch packages become outdated internally...but I'm not sure when that is. We will remove the call that is passing in positional args in D48784113, so future migration will hopefully be easier...

**Review by zou3519 — APPROVED** [review]
okey doke

**Comment by angelayi** [comment]
@zou3519 should this change be in the 2.1 release?

**Comment by zou3519** [comment]
The ideal state is if we can undo this change. If we cannot undo this change, then this should go into the branch cut so that the API is consistent
